### PR TITLE
Add disable admin menu link configuration

### PR DESCRIPTION
**Pull request by igorbp**
Previously, users could not disable the added subscriptions link. This adds a configuration option to make it easier.
 
This is an attempt to solve the issue raised in #163.

### CODE REVIEW

File: lib/solidus_subscriptions/configuration.rb [hunk 1]
```diff
@@ -4,14 +4,14 @@ module SolidusSubscriptions
...
-      :churn_buster_api_key,
+      :churn_buster_api_key, :disable_admin_menu_link
     )
 
     attr_writer(
       :success_dispatcher_class, :failure_dispatcher_class, :payment_failed_dispatcher_class,
       :out_of_stock_dispatcher, :maximum_successive_skips, :reprocessing_interval,
       :minimum_cancellation_notice, :processing_queue, :subscription_line_item_attributes,
-      :subscription_attributes, :subscribable_class,
+      :subscription_attributes, :subscribable_class
```
Comment (nirebu) [inline lib/solidus_subscriptions/configuration.rb:14]: Usually it's good to have trailing commas in order to minimise line changes when adding new elements 😉

File: lib/solidus_subscriptions/configuration.rb [hunk 2]
```diff
@@ -4,14 +4,14 @@ module SolidusSubscriptions
   class Configuration
     attr_accessor(
       :maximum_total_skips, :maximum_reprocessing_attempts, :churn_buster_account_id,
-      :churn_buster_api_key,
+      :churn_buster_api_key, :disable_admin_menu_link
```
Comment (nirebu) [inline lib/solidus_subscriptions/configuration.rb:7]: Same as below

File: lib/solidus_subscriptions/configuration.rb (hunk 1 above)
Comment (igorbp) [reply lib/solidus_subscriptions/configuration.rb:14]: Good call. Thanks, @nirebu :D

**Review by nirebu — APPROVED** [review]

File: app/overrides/views/admin_subscriptions_menu_link.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-if !Spree::Backend::Config.respond_to?(:menu_items)
+if !SolidusSubscriptions.configuration.disable_admin_menu_link && !Spree::Backend::Config.respond_to?(:menu_items)
```
Comment (aldesantis) [inline app/overrides/views/admin_subscriptions_menu_link.rb:3]: We can remove this override completely. `menu_items` is available in Solidus 2.9, which is the earliest Solidus version we currently support: https://solidus.io/security/.

**Review by aldesantis — CHANGES_REQUESTED** [review]
@igorbp I think we can remove the Deface override completely, and then I have some doubts about the overall approach you took.

Disabling the admin menu item completely and then asking the user to add it back delegates a lot of responsibility to the user, and it doesn't allow them to benefit from any updates in that area of the code.

Can't users already customize the menu item by looping through `Spree::Backend::Config.menu_items`?

**Comment by igorbp** [comment]
@aldesantis I tried to edit that through the `Spree::Backend::Config.menu_items`, but it's not that easy to find the correct moment where the extension adds their menus. During the Solidus app configuration, it doesn't have the key for it. So, it's easy to change the ones from Solidus but not from the extensions, I suppose. But I see your point, maybe we should instead provide a way of change the configuration?

**Pull request closed without merging**